src/pages/_proxied-dot-io/consul/docs/index.tsx

```diff
@@ -0,0 +1,89 @@
+import { ReactElement } from 'react'
+import { Products } from '@hashicorp/platform-product-meta'
+import ConsulIoLayout from 'layouts/_proxied-dot-io/consul'
+import { DocsPageInner, DocsPageProps } from '@hashicorp/react-docs-page'
+import productData from 'data/consul.json'
+import { isVersionedDocsEnabled } from 'lib/env-checks'
+import ProductDocsLanding from 'views/_proxied-dot-io/product-docs-landing'
+// Note: content is imported directly. Could be moved to getStaticProps.
+// ref: https://github.com/hashicorp/dev-portal/commit/cee04fb08dda81e95ebcf9e6ff08b6245872589b
+import PAGE_CONTENT from './content.json'
+// Imports below are used in getStatic functions only
+import { getStaticGenerationFunctions } from 'lib/_proxied-dot-io/get-static-generation-functions'
+import { GetStaticProps } from 'next'
+
+const product = { name: productData.name, slug: productData.slug }
+const basePath = 'docs'
+const navDataFile = `../data/${basePath}-nav-data.json`
+const localContentDir = `../content/${basePath}`
+const localPartialsDir = `../content/partials`
+const enableVersionedDocs = isVersionedDocsEnabled(productData.slug)
+
+/**
+ * Note: this route is set up solely for the `/docs` landing page.
+ * For other pages, we've switched `/docs/[[...page]].tsx`, an "optional catch-all",
+ * to `/docs/[...page].tsx`, a "catch-all" route. As mentioned in the NextJS
+ * docs, the main difference is that the latter will not match the route
+ * without parameters - ie the landing page. This allows us to avoid
+ * conflicting page files.
+ * ref: https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes
+ */
+function ConsulDocsLandingPage({
+  frontMatter,
+  currentPath,
+  navData,
+  githubFileUrl,
+  versions,
+}: DocsPageProps['staticProps']): ReactElement {
+  return (
+    <DocsPageInner
+      canonicalUrl={frontMatter.canonical_url}
+      description={frontMatter.description}
+      githubFileUrl={githubFileUrl}
+      navData={navData}
+      currentPath={currentPath}
+      pageTitle={frontMatter.page_title}
+      product={product}
+      showEditPage={false}
+      showVersionSelect={enableVersionedDocs}
+      baseRoute={basePath}
+      versions={versions}
+      algoliaConfig={productData.algoliaConfig}
+    >
+      <ProductDocsLanding
+        themeSlug={productData.slug as Products}
```

Comment:
I suppose an alternative here would be typing `productData`. However, that felt like a bit much of an undertaking for the scope of this PR.

Reply:
Ended up removing this as theming is automatically handled ✅